sources/packages/backend/apps/api/src/services/student-appeal/student-appeal.service.ts

```diff
@@ -294,12 +295,20 @@ export class StudentAppealService extends RecordDataModelService<StudentAppeal>
       .select(["studentAppeal.id"])
       .addSelect(this.buildStatusSelect(), "status")
       .innerJoin("studentAppeal.application", "application")
-      .where("application.id = :applicationId", { applicationId })
-      .andWhere("application.student.id = :studentId", { studentId })
-      .orderBy("studentAppeal.submittedDate", "DESC");
+      .where("application.id = :applicationId", { applicationId });
+
+    if (options?.studentId) {
+      query.andWhere("application.student.id = :studentId", {
+        studentId: options.studentId,
+      });
+    }
+
+    query.orderBy("studentAppeal.submittedDate", "DESC");
```

Comment:
Minor, I believe the `orderBy` can be added in the main query after the first `where`.

Reply:
I will test and try to add that in upcoming PR.